src/core/core.datasetController.js

```diff
@@ -468,6 +470,36 @@ helpers.extend(DatasetController.prototype, {
 		model.borderWidth = resolve([custom.hoverBorderWidth, dataset.hoverBorderWidth, model.borderWidth], undefined, index);
 	},
 
+	removeDatasetHoverStyle: function() {
+		var element = this.getMeta().dataset;
+
+		if (element) {
+			this.removeHoverStyle(element);
+		}
+	},
+
+	setDatasetHoverStyle: function() {
```

Comment:
I'm wondering if we want to make these public? E.g. maybe in the future we decide that a line isn't a dataset element but is more of a point element because we want to allow you to style line segments differently. I'm okay either way though

Reply:
I'm going to set these private, for completeness.